cmd/entrypoint/testutil_fake_syntheticauth_test.go

```diff
@@ -574,3 +574,158 @@ spec:
+	assert.Equal(t, "synthetic-edge-stack-auth", snap.Kubernetes.AuthServices[0].Name)
+	// In edge-stack we should only ever have 1 AuthService.
+	assert.Equal(t, 1, len(snap.Kubernetes.AuthServices))
+
+	// Even though it is the synthetic AuthService, we should have the custom timeout_ms and v3 protocol version
+	for _, authService := range snap.Kubernetes.AuthServices {
+		assert.Equal(t, int64(12345), authService.Spec.Timeout.Duration.Milliseconds())
+		assert.Equal(t, "v3", authService.Spec.ProtocolVersion)
+	}
+
+	// Check for an ext_authz cluster name matching the synthetic AuthService.
+	// the namespace for this extauthz cluster should be default (since that is the namespace of the synthetic AuthService)
+	isAuthCluster := func(c *v3cluster.Cluster) bool {
+		return strings.Contains(c.Name, "cluster_extauth_127_0_0_1_8500_default")
+	}
+
+	// Grab the next Envoy config that has an Edge Stack auth cluster on 127.0.0.1:8500
+	envoyConfig, err := f.GetEnvoyConfig(func(envoy *v3bootstrap.Bootstrap) bool {
+		return FindCluster(envoy, isAuthCluster) != nil
+	})
+	require.NoError(t, err)
+
+	// Make sure an Envoy Config containing a extauth cluster for the AuthService that was defined
+	assert.NotNil(t, envoyConfig)
+
+	t.Setenv("EDGE_STACK", "")
+}
+
+// This AuthService does not point at 127.0.0.1:8500, so despite not having protocol_version: v3, we leave it alone
+// The strict enforcement of protocol_version: v3 is only important for the AuthService that points at edge-stack
+func TestSyntheticAuthCustomAuthService(t *testing.T) {
+	t.Setenv("EDGE_STACK", "true")
+
+	f := entrypoint.RunFake(t, entrypoint.FakeConfig{EnvoyConfig: true}, nil)
+
+	err := f.UpsertYAML(`
+---
+apiVersion: getambassador.io/v2
+kind: AuthService
+metadata:
+  name: edge-stack-auth-test
+  namespace: foo
+spec:
+  auth_service: dummy-service
+  proto: "grpc"
+`)
+
+	// Use the predicate above to check that the snapshot contains the AuthService defined above
+	// The AuthService has protocol_Version: v3 so it should not be removed/replaced by the synthetic AuthService
+	// injected by syntheticauth.go
+	snap, err := f.GetSnapshot(HasAuthService("foo", "edge-stack-auth-test"))
+	assert.NoError(t, err)
+	assert.NotNil(t, snap)
+
+	assert.Equal(t, "edge-stack-auth-test", snap.Kubernetes.AuthServices[0].Name)
+	// In edge-stack we should only ever have 1 AuthService.
+	assert.Equal(t, 1, len(snap.Kubernetes.AuthServices))
+
+	for _, authService := range snap.Kubernetes.AuthServices {
+		assert.Equal(t, "dummy-service", authService.Spec.AuthService)
+	}
+
+	// Check for an ext_authz cluster name matching the provided AuthService (Http_Filters are harder to check since they always have the same name)
+	// the namespace for this extauthz cluster should be foo (since that is the namespace of the valid AuthService above)
+	isAuthCluster := func(c *v3cluster.Cluster) bool {
+		return strings.Contains(c.Name, "cluster_extauth_127_0_0_1_8500_foo")
```

Comment:
I was actually missing a call to `f.Flush()` after applying the above yaml for this test, and `cluster_extauth_127_0_0_1_8500_foo` should have been `cluster_extauth_dummy_service_foo`.

You can just switch [da11c1a](https://github.com/emissary-ingress/emissary/pull/4264/commits/da11c1ad20310086e28353964f02bda281550ae7) to [552b9aa](https://github.com/emissary-ingress/emissary/commit/552b9aa747a9147dd566c6d03a85f5aa12b2fef9) and that will resolve the test failure.

Reply:
Is this still relevant? Is it for @LukeShu